### PR TITLE
Support cloud run services

### LINKED ISSUE #612: List and view logs for Cloud Run services
Add GCP cloud run services as a Wash resource. Be able to list and delete them, and tail their logs.

API access: https://godoc.org/google.golang.org/api/run/v1

Accessing logs is likely similar to what's done for cloud functions in https://github.com/puppetlabs/wash/blob/master/plugin/gcp/cloudFunctionLog.go. We may be able to re-use some of the streaming logic.

### PR DESCRIPTION
**Pull request by ekinanp**
Resolves https://github.com/puppetlabs/wash/issues/612

Signed-off-by: Enis Inan <enis.inan@puppet.com>

### CODE REVIEW

File: plugin/gcp/cloudRunServiceLog.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	"google.golang.org/api/logging/v2"
+)
+
+type cloudRunServiceLog struct {
+	plugin.EntryBase
+	*cloudLogFile
+}
+
+func newCloudRunServiceLog(ctx context.Context, apiService cloudRunProjectAPIService, region string, serviceName string) (*cloudRunServiceLog, error) {
+	fields := []cloudLogEntryField{
```
Comment (ekinanp) [inline plugin/gcp/cloudRunServiceLog.go]: I couldn't find the cloud run service equivalent of `gcloud functions log read <function_name>`, so kind of guessed these field values based on symmetry w/ a cloud function's log.

File: plugin/gcp/cloudRunDir.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	apiService cloudRunProjectAPIService
+}
+
+func newCloudRunDir(ctx context.Context, client *http.Client, projID string) (*cloudRunDir, error) {
+	svc, err := run.NewService(context.Background(), option.WithHTTPClient(client))
+	if err != nil {
+		return nil, err
+	}
+	cr := &cloudRunDir{
+		EntryBase:  plugin.NewEntry("cloud_run"),
```
Comment (MikaelSmith) [inline plugin/gcp/cloudRunDir.go]: I guess `run` might be a little too generic. Would be symmetrical with `Cloud Storage` though.

**Review by MikaelSmith — APPROVED** [review]

File: plugin/gcp/cloudRunServiceLog.go (hunk 1 above)
Comment (MikaelSmith) [reply plugin/gcp/cloudRunServiceLog.go]: https://godoc.org/google.golang.org/api/logging/v2#LogEntry talks about the default when Severity is emptying being `DEFAULT`.

File: plugin/gcp/cloudRunDir.go (hunk 2 above)
Comment (ekinanp) [reply plugin/gcp/cloudRunDir.go]: I went with what's shown in the console. There, `Cloud Storage` is `Storage` while `Cloud Run` is `Cloud Run`.

File: plugin/gcp/cloudRunServiceLog.go (hunk 1 above)
Comment (ekinanp) [reply plugin/gcp/cloudRunServiceLog.go]: Ah thanks, updated.

**Review by MikaelSmith — APPROVED** [review]

**Pull request merged**